app/src/main/java/com/example/androidtraining/ui/product/ProductScreen.kt

```diff
@@ -0,0 +1,237 @@
+    val paddingLarge = dimensionResource(id = R.dimen.padding_large)
+    val paddingMedium = dimensionResource(id = R.dimen.padding_medium)
+    LaunchedEffect(true) {
+        viewModel.getProduct()
+    }
+
+    LaunchedEffect(successfulProductLoading) {
+        successfulProductLoading?.let {
+            if (!it) Toast.makeText(context, R.string.err_product_loading, Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    Scaffold(
+        topBar = {
+            CenterAlignedTopAppBar(
+                title = {
+                    Text(
+                        stringResource(id = R.string.item),
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis
+                    )
+                },
+                navigationIcon = {
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        Icon(
+                            imageVector = Icons.Outlined.ArrowBack,
+                            contentDescription = stringResource(id = R.string.arrow_back)
+                        )
+                    }
+                },
+                actions = {
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        BadgedBox(
+                            badge = {
+                                Badge(
+                                    contentColor = White,
+                                    containerColor = Purple,
+                                ) {
+                                    val badgeNumber = "8"
+                                    Text(
+                                        badgeNumber,
+                                        color = White
+                                    )
+                                }
+                            }) {
+                            Icon(
+                                Icons.Outlined.ShoppingCart,
+                                contentDescription = stringResource(id = R.string.shopping_cart)
+                            )
+                        }
+                    }
+                }
+            )
+        },
+        content = { paddingValues ->
+            viewModel.productResponse.value?.getOrNull()?.let { item ->
+                Column(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .paint(
+                            painterResource(id = R.drawable.product_background),
+                            contentScale = ContentScale.FillWidth
+                        )
+                        .padding(horizontal = paddingLarge)
+                        .padding(top = paddingValues.calculateTopPadding())
+                ) {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .clip(RoundedCornerShape(16.dp))
+                    ) {
+                        Row(
```

Comment:
No need for this `Row` as it wraps only a single element. The image should be able to `fillMaxWidth` and `clip` itself in its own modifier.